lib/matplotlib/scale.py

```diff
@@ -69,7 +69,8 @@ def set_default_locators_and_formatters(self, axis):
         axis.set_major_formatter(ScalarFormatter())
         axis.set_minor_formatter(NullFormatter())
         # update the minor locator for x and y axis based on rcParams
-        if rcParams['xtick.minor.visible']:
+        if axis.axis_name in list("xy") and \
+            rcParams['{}tick.minor.visible'.format(axis.axis_name)]:
```

Comment:
IMO this would be slightly more readable:
~~~
if (axis.axis_name == 'x' and rcParams['xtick.minor.visible']
    or axis.axis_name == 'y' and rcParams['ytick.minor.visible']):
~~~